### PR TITLE
notifications: Correctly stringify pmUsers to fix navigation to group…

### PR DESCRIPTION
**Pull request by chrisbobbe**
… PMs.

Navigation to a group PM on pressing a notification was broken
because pmUsers was incorrectly stringified in
GroupPm.getPmUsersString. Fix, and add a test for it.

### CODE REVIEW

File: android/app/src/test/java/com/zulipmobile/notifications/FcmMessageTest.kt [hunk 1]
```diff
@@ -52,6 +52,19 @@ class FcmMessageTest : FcmMessageTestBase() {
     }
 }
 
+class RecipientTest : FcmMessageTestBase() {
+    @Test
+    fun `GroupPm#getPmUsersString gives the correct string`() {
+        expect.that(Recipient.GroupPm(pmUsers = setOf(123, 234, 345)).getPmUsersString())
+            .isEqualTo("123, 234, 345")
```
Comment (gnprice) [inline android/app/src/test/java/com/zulipmobile/notifications/FcmMessageTest.kt]: Let's eliminate the spaces too.

If this already is enough that the JS code understands it, that's good :) -- but the format used elsewhere is purely comma-separated, so may as well make it consistent.

File: android/app/src/test/java/com/zulipmobile/notifications/FcmMessageTest.kt [hunk 2]
```diff
@@ -52,6 +52,19 @@ class FcmMessageTest : FcmMessageTestBase() {
...
+    }
+    @Test
+    fun `GroupPm#getPmUsersString correctly reorders user ids`() {
+        expect.that(Recipient.GroupPm(pmUsers = setOf(234, 123, 345)).getPmUsersString())
+            .isEqualTo("123, 234, 345")
+    }
```
Comment (gnprice) [inline android/app/src/test/java/com/zulipmobile/notifications/FcmMessageTest.kt]: This test can be made a bit stronger by testing that e.g. 123 sorts after 23 -- i.e. the sorting is of the numbers, and not a lexicographic sort of their ASCII representations.

**Comment by chrisbobbe** [comment]
OK, I've pushed these changes. :)

**Comment by chrisbobbe** [comment]
I'm resolving merge conflicts and clarifying a commit message for this now. ("Incorrectly stringified" is vague; I'm tracking down what exactly the output was, which I should have done in the first place.)

**Comment by gnprice** [comment]
Merged! Thanks again. Just made small tweaks to the commit messages.

Evidently I dropped this PR back in March, oops. >_> Looks like it probably got lost in the chaos of all suddenly switching to work from home, early in the pandemic. I ran across the bug today, it looked familiar, and I found this PR again.

**Pull request merged**